src/service/transaction.service.ts

```diff
@@ -9,13 +9,14 @@ import { DESCRIPTION, REFERENCE } from "../config";
 import { getApplicationData } from "../utils/application.data";
 import { ApplicationData } from "../model";
 import { makeApiCallWithRetry } from "./retry.handler.service";
-import { EntityNameKey } from "../model/data.types.model";
+import { EntityNameKey, EntityNumberKey } from "../model/data.types.model";
 
 export const postTransaction = async (req: Request, session: Session): Promise<string> => {
   const applicationData: ApplicationData = getApplicationData(session);
   const companyName = applicationData[EntityNameKey];
+  const companyNumber = applicationData[EntityNumberKey];
 
-  const transaction: Transaction = { reference: REFERENCE, companyName, description: DESCRIPTION };
+  const transaction: Transaction = { reference: REFERENCE, companyName, companyNumber, description: DESCRIPTION };
```

Comment:
Shared with registration - should companyNumber not be optional? 
May be adding undefined to transaction object

Reply:
instead 19:
const transaction: Transaction = companyNumber === undefined? { reference: REFERENCE, companyName, description: DESCRIPTION } : { reference: REFERENCE, companyName, companyNumber, description: DESCRIPTION };